[fix] Independent EBIC detector might not have MD_POS

pieleric opened the pull request:
The MD_POS needs to be copied from the scanner metadata. However, the
acquisition restart function could easily stop the scanner *just* before
receiving the data (and metadata).
=> Wait explicitly for the scanner data before restarting the
acquisition.

Also fix a few issues in the test cases and simulator.

pieleric committed:
[fix] Independent EBIC detector might not have MD_POS

The MD_POS needs to be copied from the scanner metadata. However, the
acquisition restart function could easily stop the scanner *just* before
receiving the data (and metadata).
=> Wait explicitly for the scanner data before restarting the
acquisition.

tepals commented on src/odemis/acq/stream/_helper.py:
```diff
@@ -1550,6 +1551,9 @@ def _updateAcquisitionTime(self) -> None:
         """
         # Override standard method (on LiveStream), which checks whether the acquisition time has
         # changed "a lot", and in such case stop and restart the acquisition.
+        # Here, we would need to restart both the scanner and detector, which is more complex...
+        # especially because this function is called just after active is set, which would cause an
+        # immediate restart.
```
This comment raises more questions for me than that it explains anything. For instance why would the scanner and detector need to be restarted, and compared to what is it more complex?

tepals commented on src/odemis/acq/stream/_helper.py:
```diff
@@ -1611,22 +1619,29 @@ def _onNewData(self, dataflow: model.DataFlow, data: model.DataArray) -> None:
             with self._acq_start_lock:
                 if self.is_active.value:
+                    # Make sure that the emitter data is received before restarting the acquisition,
+                    # otherwise, we might its metadata.
```
```suggestion
                    # otherwise, we might miss its metadata.
```

tepals commented on src/odemis/acq/stream/_helper.py:
```diff
@@ -1611,22 +1619,29 @@ def _onNewData(self, dataflow: model.DataFlow, data: model.DataArray) -> None:
-    def _restart_acquisition(self):
+    def _restart_acquisition(self, expect_emt_data: bool):
         """
         Stops and immediately restarts the acquisition.
```
the "immediately" is now technically not true anymore

tmoerkerken commented on src/odemis/acq/test/stream_test.py:
```diff
@@ -4185,7 +4185,7 @@ def test_ebic_live_stream(self):
         ebics = stream.IndependentEBICStream("test ebic", self.ebic, self.ebic.data,
                                              self.ebeam, self.sed.data,
-                                             detvas={"dwellTime"})
+                                             emtvas={"dwellTime"})
```
I don't see how these changes to this test are related to the PR topic (fixing setting of metadata). Would have expected a test where we actually check the metadata.

pieleric replied:
You're right. It's actually a fix for the test cases. I found the test case were broken when running them after the changes... but it's not directly related. I can put these changes in a separate commit.

pieleric committed:
[fix] SPARC IndependentDetector test case failure

Minor fixes to make the IndependentDetector test cases use the IndependentEBICStream
properly, and adjust the simulator so that it works in the test case.
